presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/utils/delegates/EcoDisposableProperty.kt

```diff
@@ -0,0 +1,15 @@
+package com.exallium.mvvmcleanexample.presentation.utils.delegates
+
+import io.reactivex.disposables.Disposable
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class EcoDisposableProperty(private var disposable: Disposable?) : ReadWriteProperty<Any?, Disposable?> {
```

Comment:
Can maybe do something with operator overloading or infix operators, but that's new territory for me. Will try.